### PR TITLE
Rtos

### PR DESCRIPTION
**Pull request by nessnah198**
timing optimizations

### CODE REVIEW

File: Src/Filter.c [hunk 1]
```diff
@@ -114,7 +118,8 @@ void StartsensorFilterTask(void const * arguments)
     // Send mail
     //  osMailPut(analys_mailbox, lowpass_data_acc_x);
     osMailPut(sensorFilter_mailbox, &complement_data);
-    vTaskDelayUntil(&last_task_start,MAIN_FREQUENCY); 
+    HAL_GPIO_WritePin(GPIOA, GPIO_PIN_8, GPIO_PIN_RESET);
```
Comment (rlindsberg) [inline Src/Filter.c]: Here also..

File: Src/analys.c [hunk 2]
```diff
@@ -62,6 +62,7 @@ void StartAnalysTask(void const * argument)
   /* Thread */
   while(1){
     
+    HAL_GPIO_WritePin(GPIOC, GPIO_PIN_15, GPIO_PIN_SET);
```
Comment (rlindsberg) [inline Src/analys.c]: And here..

File: Src/analys.c [hunk 3]
```diff
@@ -107,8 +108,8 @@ void StartAnalysTask(void const * argument)
     HAL_UART_Transmit(&huart3,(uint8_t*)&val, sizeof(float)*ctr, 4);
     huart3.State = HAL_UART_STATE_READY;
 
-    
+    HAL_GPIO_WritePin(GPIOC, GPIO_PIN_15, GPIO_PIN_RESET);
```
Comment (rlindsberg) [inline Src/analys.c]: And here..

File: Src/Filter.c [hunk 4]
```diff
@@ -87,6 +88,9 @@ void StartsensorFilterTask(void const * arguments)
   
   while(1)
   {
+    
+    HAL_GPIO_WritePin(GPIOA, GPIO_PIN_8, GPIO_PIN_SET);
```
Comment (rlindsberg) [inline Src/Filter.c]: Why would you want to write pin here?

File: Src/controlThread.c [hunk 5]
```diff
@@ -94,8 +98,11 @@ void StartControlTask(void const * arguments)
     /* Watchdog check back */
     gMainloopWDCheckback++;
 
+    HAL_GPIO_WritePin(GPIOA, GPIO_PIN_6, GPIO_PIN_RESET);
+
+    
     /* Delay the task to a fixed time to ensure constant execution frequncy */
-    vTaskDelayUntil(&last_task_start,CONTROLL_FREQUENCY); 
+    vTaskDelayUntil(&last_task_start,HYPERPERIOD/2);
```
Comment (rlindsberg) [inline Src/controlThread.c]: Why change from execution frequency from 2 to 10?

File: Src/Filter.c (hunk 1 above)
Comment (nessnah198) [reply Src/Filter.c]: pins for analyzing threads

File: Src/analys.c (hunk 2 above)
Comment (nessnah198) [reply Src/analys.c]: same here

File: Src/analys.c (hunk 3 above)
Comment (nessnah198) [reply Src/analys.c]: and here. analyzing threads

File: README.md [hunk 6]
```diff
@@ -1,6 +1,13 @@
 # Quadcopter-Flight-Controller
 ## README / INSTRUCTIONS
 ## Analys
+### Timing and time analys
+```
+Analysthread:   8ms   execution time    runs 1 time  in 20ms hyperperiod  
+Controlthread:  4ms   execution time    runs 2 times in 20ms hyperperiod  
+Filterthread:   160us execution time    runs 2 times in 20ms hyperperiod  
+Pwminthread:    1ms   execution time    runs 2 times in 20ms hyperperiod
```
Comment (rlindsberg) [inline README.md]: 5 times*

**Comment by rlindsberg** [comment]
Wait for Johan to update readme

**Pull request merged**